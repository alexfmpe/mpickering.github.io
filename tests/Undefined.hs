class AwsType a where
    toText :: a -> b


    {-# MINIMAL toText #-}

class Minimal a where
  toText :: a -> b
  {-# MINIMAL decimal, hexadecimal, realFloat, scientific #-}

class Minimal a where
  toText :: a -> b
  {-# MINIMAL shape, (maskedIndex | maskedLinearIndex) #-}

class Minimal a where
  toText :: a -> b
  {-# MINIMAL (toSample | toSamples) #-}

