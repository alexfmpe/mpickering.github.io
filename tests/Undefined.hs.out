class AwsType a where
    toText :: a -> b


    {-# MINIMAL toText #-}

class Minimal a where
  toText :: a -> b
  {-# MINIMAL decimal, hexadecimal  realFloat  scientific #-}

class Minimal a where
  toText :: a -> b
  {-# MINIMAL shape,  maskedIndex | maskedLinearIndex  #-}

class Minimal a where
  toText :: a -> b
  {-# MINIMAL  toSample | toSamples  #-}


==============
tests/examples/Undefined.hs
==============
lengths:(361,361)

==============

({ tests/examples/Undefined.hs:1:1 }
 Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEofPos),DP (2,0))])
 (HsModule 
  (Nothing) 
  (Nothing) 
  [] 
  [
   ({ tests/examples/Undefined.hs:(1,1)-(5,26) }
    Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])
    (TyClD 
     (ClassDecl 
      ({ <no location info> }
       Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
       []) 
      ({ tests/examples/Undefined.hs:1:7-13 }
       Just (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])
       (Unqual {OccName: AwsType})) 
      (HsQTvs 
       [] 
       [
        ({ tests/examples/Undefined.hs:1:15 }
         Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
         (UserTyVar 
          (Unqual {OccName: a})))]) 
      [] 
      [
       ({ tests/examples/Undefined.hs:2:5-20 }
        Just (Ann (DP (1,5)) (ColDelta 5) DP (1,5) [] [((G AnnDcolon),DP (0,1))])
        (TypeSig 
         [
          ({ tests/examples/Undefined.hs:2:5-10 }
           Just (Ann (DP (0,0)) (ColDelta 5) DP (0,0) [] [((G AnnVal),DP (0,0))])
           (Unqual {OccName: toText}))] 
         ({ tests/examples/Undefined.hs:2:15-20 }
          Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [])
          (HsForAllTy 
           (Implicit) 
           (Nothing) 
           (HsQTvs 
            [] 
            []) 
           ({ <no location info> }
            Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
            []) 
           ({ tests/examples/Undefined.hs:2:15-20 }
            Just (Ann (DP (0,0)) (ColDelta 15) DP (0,0) [] [((G AnnRarrow),DP (0,1))])
            (HsFunTy 
             ({ tests/examples/Undefined.hs:2:15 }
              Just (Ann (DP (0,0)) (ColDelta 15) DP (0,0) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: a}))) 
             ({ tests/examples/Undefined.hs:2:20 }
              Just (Ann (DP (0,1)) (ColDelta 20) DP (0,1) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: b}))))))) 
         (PlaceHolder))),
       ({ tests/examples/Undefined.hs:5:5-26 }
        Just (Ann (DP (3,5)) (ColDelta 5) DP (3,5) [] [((G AnnOpen),DP (0,0)),((G AnnClose),DP (0,1))])
        (MinimalSig "{-# MINIMAL" 
         (Var 
          ({ tests/examples/Undefined.hs:5:17-22 }
           Just (Ann (DP (0,1)) (ColDelta 17) DP (0,1) [] [((G AnnVal),DP (0,0))])
           (Unqual {OccName: toText})))))] {Bag(Located (HsBind RdrName)): 
      []} 
      [] 
      [] 
      [] 
      (PlaceHolder)))),
   ({ tests/examples/Undefined.hs:(7,1)-(9,61) }
    Just (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])
    (TyClD 
     (ClassDecl 
      ({ <no location info> }
       Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
       []) 
      ({ tests/examples/Undefined.hs:7:7-13 }
       Just (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])
       (Unqual {OccName: Minimal})) 
      (HsQTvs 
       [] 
       [
        ({ tests/examples/Undefined.hs:7:15 }
         Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
         (UserTyVar 
          (Unqual {OccName: a})))]) 
      [] 
      [
       ({ tests/examples/Undefined.hs:8:3-18 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnDcolon),DP (0,1))])
        (TypeSig 
         [
          ({ tests/examples/Undefined.hs:8:3-8 }
           Just (Ann (DP (0,0)) (ColDelta 3) DP (0,0) [] [((G AnnVal),DP (0,0))])
           (Unqual {OccName: toText}))] 
         ({ tests/examples/Undefined.hs:8:13-18 }
          Just (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [])
          (HsForAllTy 
           (Implicit) 
           (Nothing) 
           (HsQTvs 
            [] 
            []) 
           ({ <no location info> }
            Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
            []) 
           ({ tests/examples/Undefined.hs:8:13-18 }
            Just (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnRarrow),DP (0,1))])
            (HsFunTy 
             ({ tests/examples/Undefined.hs:8:13 }
              Just (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: a}))) 
             ({ tests/examples/Undefined.hs:8:18 }
              Just (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: b}))))))) 
         (PlaceHolder))),
       ({ tests/examples/Undefined.hs:9:3-61 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnOpen),DP (0,0)),((G AnnComma),DP (0,0)),((G AnnClose),DP (0,1))])
        (MinimalSig "{-# MINIMAL" 
         (And 
          [
           (Var 
            ({ tests/examples/Undefined.hs:9:15-21 }
             Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (Unqual {OccName: decimal}))),
           (And 
            [
             (Var 
              ({ tests/examples/Undefined.hs:9:24-34 }
               Just (Ann (DP (0,1)) (ColDelta 24) DP (0,1) [] [((G AnnVal),DP (0,0))])
               (Unqual {OccName: hexadecimal}))),
             (And 
              [
               (Var 
                ({ tests/examples/Undefined.hs:9:37-45 }
                 Just (Ann (DP (0,2)) (ColDelta 37) DP (0,2) [] [((G AnnVal),DP (0,0))])
                 (Unqual {OccName: realFloat}))),
               (Var 
                ({ tests/examples/Undefined.hs:9:48-57 }
                 Just (Ann (DP (0,2)) (ColDelta 48) DP (0,2) [] [((G AnnVal),DP (0,0))])
                 (Unqual {OccName: scientific})))])])])))] {Bag(Located (HsBind RdrName)): 
      []} 
      [] 
      [] 
      [] 
      (PlaceHolder)))),
   ({ tests/examples/Undefined.hs:(11,1)-(13,58) }
    Just (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])
    (TyClD 
     (ClassDecl 
      ({ <no location info> }
       Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
       []) 
      ({ tests/examples/Undefined.hs:11:7-13 }
       Just (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])
       (Unqual {OccName: Minimal})) 
      (HsQTvs 
       [] 
       [
        ({ tests/examples/Undefined.hs:11:15 }
         Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
         (UserTyVar 
          (Unqual {OccName: a})))]) 
      [] 
      [
       ({ tests/examples/Undefined.hs:12:3-18 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnDcolon),DP (0,1))])
        (TypeSig 
         [
          ({ tests/examples/Undefined.hs:12:3-8 }
           Just (Ann (DP (0,0)) (ColDelta 3) DP (0,0) [] [((G AnnVal),DP (0,0))])
           (Unqual {OccName: toText}))] 
         ({ tests/examples/Undefined.hs:12:13-18 }
          Just (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [])
          (HsForAllTy 
           (Implicit) 
           (Nothing) 
           (HsQTvs 
            [] 
            []) 
           ({ <no location info> }
            Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
            []) 
           ({ tests/examples/Undefined.hs:12:13-18 }
            Just (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnRarrow),DP (0,1))])
            (HsFunTy 
             ({ tests/examples/Undefined.hs:12:13 }
              Just (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: a}))) 
             ({ tests/examples/Undefined.hs:12:18 }
              Just (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: b}))))))) 
         (PlaceHolder))),
       ({ tests/examples/Undefined.hs:13:3-58 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnOpen),DP (0,0)),((G AnnComma),DP (0,0)),((G AnnVbar),DP (0,1)),((G AnnClose),DP (0,2))])
        (MinimalSig "{-# MINIMAL" 
         (And 
          [
           (Var 
            ({ tests/examples/Undefined.hs:13:15-19 }
             Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (Unqual {OccName: shape}))),
           (Or 
            [
             (Var 
              ({ tests/examples/Undefined.hs:13:23-33 }
               Just (Ann (DP (0,2)) (ColDelta 23) DP (0,2) [] [((G AnnVal),DP (0,0))])
               (Unqual {OccName: maskedIndex}))),
             (Var 
              ({ tests/examples/Undefined.hs:13:37-53 }
               Just (Ann (DP (0,1)) (ColDelta 37) DP (0,1) [] [((G AnnVal),DP (0,0))])
               (Unqual {OccName: maskedLinearIndex})))])])))] {Bag(Located (HsBind RdrName)): 
      []} 
      [] 
      [] 
      [] 
      (PlaceHolder)))),
   ({ tests/examples/Undefined.hs:(15,1)-(17,40) }
    Just (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])
    (TyClD 
     (ClassDecl 
      ({ <no location info> }
       Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
       []) 
      ({ tests/examples/Undefined.hs:15:7-13 }
       Just (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])
       (Unqual {OccName: Minimal})) 
      (HsQTvs 
       [] 
       [
        ({ tests/examples/Undefined.hs:15:15 }
         Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
         (UserTyVar 
          (Unqual {OccName: a})))]) 
      [] 
      [
       ({ tests/examples/Undefined.hs:16:3-18 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnDcolon),DP (0,1))])
        (TypeSig 
         [
          ({ tests/examples/Undefined.hs:16:3-8 }
           Just (Ann (DP (0,0)) (ColDelta 3) DP (0,0) [] [((G AnnVal),DP (0,0))])
           (Unqual {OccName: toText}))] 
         ({ tests/examples/Undefined.hs:16:13-18 }
          Just (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [])
          (HsForAllTy 
           (Implicit) 
           (Nothing) 
           (HsQTvs 
            [] 
            []) 
           ({ <no location info> }
            Just (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] [])
            []) 
           ({ tests/examples/Undefined.hs:16:13-18 }
            Just (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnRarrow),DP (0,1))])
            (HsFunTy 
             ({ tests/examples/Undefined.hs:16:13 }
              Just (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: a}))) 
             ({ tests/examples/Undefined.hs:16:18 }
              Just (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])
              (HsTyVar 
               (Unqual {OccName: b}))))))) 
         (PlaceHolder))),
       ({ tests/examples/Undefined.hs:17:3-40 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnOpen),DP (0,0)),((G AnnVbar),DP (0,1)),((G AnnClose),DP (0,2))])
        (MinimalSig "{-# MINIMAL" 
         (Or 
          [
           (Var 
            ({ tests/examples/Undefined.hs:17:16-23 }
             Just (Ann (DP (0,2)) (ColDelta 16) DP (0,2) [] [((G AnnVal),DP (0,0))])
             (Unqual {OccName: toSample}))),
           (Var 
            ({ tests/examples/Undefined.hs:17:27-35 }
             Just (Ann (DP (0,1)) (ColDelta 27) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (Unqual {OccName: toSamples})))])))] {Bag(Located (HsBind RdrName)): 
      []} 
      [] 
      [] 
      [] 
      (PlaceHolder))))] 
  (Nothing) 
  (Nothing)))
==============
[(AnnKey tests/examples/Undefined.hs:1:1 CN "HsModule",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEofPos),DP (2,0))])),
 (AnnKey tests/examples/Undefined.hs:(1,1)-(5,26) CN "TyClD",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:1:7-13 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:1:15 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:2:5-10 CN "Unqual",
  (Ann (DP (0,0)) (ColDelta 5) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:2:5-20 CN "TypeSig",
  (Ann (DP (1,5)) (ColDelta 5) DP (1,5) [] [((G AnnDcolon),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:2:15 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 15) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:2:15-20 CN "HsForAllTy",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [])),
 (AnnKey tests/examples/Undefined.hs:2:15-20 CN "HsFunTy",
  (Ann (DP (0,0)) (ColDelta 15) DP (0,0) [] [((G AnnRarrow),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:2:20 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 20) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:5:5-26 CN "MinimalSig",
  (Ann (DP (3,5)) (ColDelta 5) DP (3,5) [] [((G AnnOpen),DP (0,0)),((G AnnClose),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:5:17-22 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 17) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:(7,1)-(9,61) CN "TyClD",
  (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:7:7-13 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:7:15 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:8:3-8 CN "Unqual",
  (Ann (DP (0,0)) (ColDelta 3) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:8:3-18 CN "TypeSig",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnDcolon),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:8:13 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:8:13-18 CN "HsForAllTy",
  (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [])),
 (AnnKey tests/examples/Undefined.hs:8:13-18 CN "HsFunTy",
  (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnRarrow),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:8:18 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:9:3-61 CN "MinimalSig",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnOpen),DP (0,0)),((G AnnComma),DP (0,0)),((G AnnClose),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:9:15-21 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:9:24-34 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 24) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:9:37-45 CN "Unqual",
  (Ann (DP (0,2)) (ColDelta 37) DP (0,2) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:9:48-57 CN "Unqual",
  (Ann (DP (0,2)) (ColDelta 48) DP (0,2) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:(11,1)-(13,58) CN "TyClD",
  (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:11:7-13 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:11:15 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:12:3-8 CN "Unqual",
  (Ann (DP (0,0)) (ColDelta 3) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:12:3-18 CN "TypeSig",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnDcolon),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:12:13 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:12:13-18 CN "HsForAllTy",
  (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [])),
 (AnnKey tests/examples/Undefined.hs:12:13-18 CN "HsFunTy",
  (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnRarrow),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:12:18 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:13:3-58 CN "MinimalSig",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnOpen),DP (0,0)),((G AnnComma),DP (0,0)),((G AnnVbar),DP (0,1)),((G AnnClose),DP (0,2))])),
 (AnnKey tests/examples/Undefined.hs:13:15-19 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:13:23-33 CN "Unqual",
  (Ann (DP (0,2)) (ColDelta 23) DP (0,2) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:13:37-53 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 37) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:(15,1)-(17,40) CN "TyClD",
  (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:15:7-13 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:15:15 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:16:3-8 CN "Unqual",
  (Ann (DP (0,0)) (ColDelta 3) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:16:3-18 CN "TypeSig",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnDcolon),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:16:13 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:16:13-18 CN "HsForAllTy",
  (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [])),
 (AnnKey tests/examples/Undefined.hs:16:13-18 CN "HsFunTy",
  (Ann (DP (0,0)) (ColDelta 13) DP (0,0) [] [((G AnnRarrow),DP (0,1))])),
 (AnnKey tests/examples/Undefined.hs:16:18 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:17:3-40 CN "MinimalSig",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnOpen),DP (0,0)),((G AnnVbar),DP (0,1)),((G AnnClose),DP (0,2))])),
 (AnnKey tests/examples/Undefined.hs:17:16-23 CN "Unqual",
  (Ann (DP (0,2)) (ColDelta 16) DP (0,2) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/Undefined.hs:17:27-35 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 27) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey <no location info> CN "[]",
  (Ann (DP (-1,0)) (ColDelta 0) DP (-1,0) [] []))]
==============
([((tests/examples/Undefined.hs:(1,1)-(5,26), AnnClass),
   [tests/examples/Undefined.hs:1:1-5]),
  ((tests/examples/Undefined.hs:(1,1)-(5,26), AnnSemi),
   [tests/examples/Undefined.hs:7:1]),
  ((tests/examples/Undefined.hs:(1,1)-(5,26), AnnWhere),
   [tests/examples/Undefined.hs:1:17-21]),
  ((tests/examples/Undefined.hs:2:5-20, AnnDcolon),
   [tests/examples/Undefined.hs:2:12-13]),
  ((tests/examples/Undefined.hs:2:5-20, AnnSemi),
   [tests/examples/Undefined.hs:5:5]),
  ((tests/examples/Undefined.hs:2:15-20, AnnRarrow),
   [tests/examples/Undefined.hs:2:17-18]),
  ((tests/examples/Undefined.hs:5:5-26, AnnClose),
   [tests/examples/Undefined.hs:5:24-26]),
  ((tests/examples/Undefined.hs:5:5-26, AnnOpen),
   [tests/examples/Undefined.hs:5:5-15]),
  ((tests/examples/Undefined.hs:(7,1)-(9,61), AnnClass),
   [tests/examples/Undefined.hs:7:1-5]),
  ((tests/examples/Undefined.hs:(7,1)-(9,61), AnnSemi),
   [tests/examples/Undefined.hs:11:1]),
  ((tests/examples/Undefined.hs:(7,1)-(9,61), AnnWhere),
   [tests/examples/Undefined.hs:7:17-21]),
  ((tests/examples/Undefined.hs:8:3-18, AnnDcolon),
   [tests/examples/Undefined.hs:8:10-11]),
  ((tests/examples/Undefined.hs:8:3-18, AnnSemi),
   [tests/examples/Undefined.hs:9:3]),
  ((tests/examples/Undefined.hs:8:13-18, AnnRarrow),
   [tests/examples/Undefined.hs:8:15-16]),
  ((tests/examples/Undefined.hs:9:3-61, AnnClose),
   [tests/examples/Undefined.hs:9:59-61]),
  ((tests/examples/Undefined.hs:9:3-61, AnnComma),
   [tests/examples/Undefined.hs:9:22,
    tests/examples/Undefined.hs:9:35,
    tests/examples/Undefined.hs:9:46]),
  ((tests/examples/Undefined.hs:9:3-61, AnnOpen),
   [tests/examples/Undefined.hs:9:3-13]),
  ((tests/examples/Undefined.hs:(11,1)-(13,58), AnnClass),
   [tests/examples/Undefined.hs:11:1-5]),
  ((tests/examples/Undefined.hs:(11,1)-(13,58), AnnSemi),
   [tests/examples/Undefined.hs:15:1]),
  ((tests/examples/Undefined.hs:(11,1)-(13,58), AnnWhere),
   [tests/examples/Undefined.hs:11:17-21]),
  ((tests/examples/Undefined.hs:12:3-18, AnnDcolon),
   [tests/examples/Undefined.hs:12:10-11]),
  ((tests/examples/Undefined.hs:12:3-18, AnnSemi),
   [tests/examples/Undefined.hs:13:3]),
  ((tests/examples/Undefined.hs:12:13-18, AnnRarrow),
   [tests/examples/Undefined.hs:12:15-16]),
  ((tests/examples/Undefined.hs:13:3-58, AnnClose),
   [tests/examples/Undefined.hs:13:56-58]),
  ((tests/examples/Undefined.hs:13:3-58, AnnCloseP),
   [tests/examples/Undefined.hs:13:54]),
  ((tests/examples/Undefined.hs:13:3-58, AnnComma),
   [tests/examples/Undefined.hs:13:20]),
  ((tests/examples/Undefined.hs:13:3-58, AnnOpen),
   [tests/examples/Undefined.hs:13:3-13]),
  ((tests/examples/Undefined.hs:13:3-58, AnnOpenP),
   [tests/examples/Undefined.hs:13:22]),
  ((tests/examples/Undefined.hs:13:3-58, AnnVbar),
   [tests/examples/Undefined.hs:13:35]),
  ((tests/examples/Undefined.hs:(15,1)-(17,40), AnnClass),
   [tests/examples/Undefined.hs:15:1-5]),
  ((tests/examples/Undefined.hs:(15,1)-(17,40), AnnSemi),
   [tests/examples/Undefined.hs:19:1]),
  ((tests/examples/Undefined.hs:(15,1)-(17,40), AnnWhere),
   [tests/examples/Undefined.hs:15:17-21]),
  ((tests/examples/Undefined.hs:16:3-18, AnnDcolon),
   [tests/examples/Undefined.hs:16:10-11]),
  ((tests/examples/Undefined.hs:16:3-18, AnnSemi),
   [tests/examples/Undefined.hs:17:3]),
  ((tests/examples/Undefined.hs:16:13-18, AnnRarrow),
   [tests/examples/Undefined.hs:16:15-16]),
  ((tests/examples/Undefined.hs:17:3-40, AnnClose),
   [tests/examples/Undefined.hs:17:38-40]),
  ((tests/examples/Undefined.hs:17:3-40, AnnCloseP),
   [tests/examples/Undefined.hs:17:36]),
  ((tests/examples/Undefined.hs:17:3-40, AnnOpen),
   [tests/examples/Undefined.hs:17:3-13]),
  ((tests/examples/Undefined.hs:17:3-40, AnnOpenP),
   [tests/examples/Undefined.hs:17:15]),
  ((tests/examples/Undefined.hs:17:3-40, AnnVbar),
   [tests/examples/Undefined.hs:17:25]),
  ((<no location info>, AnnEofPos),
   [tests/examples/Undefined.hs:19:1])],
 [(<no location info>, [])])